SparseGrids/tsgEnumerates.hpp

```diff
@@ -39,81 +39,460 @@
 
 #include "TasmanianConfig.hpp"
 
+//! \file tsgEnumerates.hpp
+//! \brief Omnipresent enumerate types.
+//! \author Miroslav Stoyanov
+//! \ingroup TasmanianEnumerates
+//!
+//! Contains the enumerate types that are used throughout Tasmanian.
+//! This header is included directly or transitively in every other
+//! file of the project including \b TasmanianSparseGrids.hpp
+
+//! \defgroup TasmanianEnumerates Enumerate types.
+//!
+//! \par Enumerate types
+//! The enumerates are used as input to many Tasmanian functions
+//! in both internal and external API.
+//!
+//! \internal
+//! \par Hardcoded constants
+//! Hardcoding constants is a bad coding practice and should be avoided,
+//! but numerical algorithms depend on many small tweaking parameters which
+//! encumbers the user.
```

Comment:
Many calls to Tasmanian would require 2 - 3 extra variables which will have different meaning for different rules. For example, Gauss-Legendre nodes are computed by a different algorithm than Leja nodes, both algorithms require tolerance and maximum number of iterations (when the number of nodes is large), but why should the user care? In vast majority of cases this will not happen.

Reply:
Ah, got it!